src/Filter/Generic/IEnumerable`1Extensions.cs

```diff
@@ -46,60 +46,60 @@ public static class IEnumerable_1Extensions
                         {
                             queryableValue = queryableValue.Contains(validValuePropertyName, (IEnumerable)filterPropertyValue);
                         }
-                        else if (filterProperty.PropertyType.IsGenericType &&
-                            (
-                            typeof(IRange<>).IsAssignableFrom(filterProperty.PropertyType.GetGenericTypeDefinition()) ||
-                            typeof(Range<>).IsAssignableFrom(filterProperty.PropertyType.GetGenericTypeDefinition())))
+                        else if (typeof(IRange<>).IsAssignableFrom(filterProperty.PropertyType.GetGenericTypeDefinition()) ||
```

Comment:
Update this logic so everything gets funneled to `IRange<>`. Concrete type of filter property-value is no longer important as long as it implements the interface.